src/forms/medication-dispense-review.component.tsx

```diff
@@ -351,136 +355,191 @@ const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
-                    ],
-                  },
-                ],
-              });
-            }}
-          />
-
           <ResponsiveWrapper>
-            <ComboBox
-              id="dosingUnits"
+            <Toggle
+              toggled={isFreeTextDosage}
               disabled={!userCanModify || !allowEditing}
-              items={drugDosingUnits}
-              titleText={t('doseUnit', 'Dose unit')}
-              itemToString={(item) => item?.text}
-              initialSelectedItem={{
-                id: medicationDispense.dosageInstruction[0].doseAndRate[0].doseQuantity?.code,
-                text: medicationDispense.dosageInstruction[0].doseAndRate[0].doseQuantity?.unit,
-              }}
-              onChange={({ selectedItem }) => {
-                updateMedicationDispense({
-                  dosageInstruction: [
-                    {
-                      ...medicationDispense.dosageInstruction[0],
-                      doseAndRate: [
-                        {
-                          doseQuantity: {
-                            // note that we specifically recreate doseQuantity to overwrite any unit or system properties that may have been set
-                            value: medicationDispense.dosageInstruction[0].doseAndRate[0].doseQuantity?.value,
-                            code: selectedItem?.id,
+              id="isFreeTextToggled"
+              size={'sm'}
+              labelText={t('freeTextDosage', 'Free text dosage')}
+              onToggle={(value) => {
+                setIsFreeTextDosage(value);
+                // clear out the dose, route and frequency if changed to free text dosage
+                if (value) {
+                  updateMedicationDispense({
+                    dosageInstruction: [
+                      {
+                        ...medicationDispense.dosageInstruction[0],
+                        doseAndRate: [
+                          {
+                            doseQuantity: {
+                              value: null,
+                              code: null,
+                            },
+                          },
+                        ],
+                        route: {
+                          coding: [
+                            {
+                              code: null,
+                            },
+                          ],
+                        },
+                        timing: {
+                          ...medicationDispense.dosageInstruction[0].timing,
+                          code: {
+                            coding: [
+                              {
+                                code: null,
+                              },
+                            ],
                           },
                         },
-                      ],
-                    },
-                  ],
-                });
+                      },
+                    ],
+                  });
+                }
```

Comment:
```suggestion
                } else {
                  // clear out the text field if changed from free text to coded dosage
                  updateMedicationDispense({
                    dosageInstruction: [
                      {
                        ...medicationDispense.dosageInstruction[0],
                        text: undefined,
                      },
                    ],
                  });
```

Before: Switching from free-text to coded left stale text, causing confusion and potential validation issues.

After: Clears the text field when switching to coded mode, keeping the form state consistent.